services/ops/TopicWatcher/topic_watcher/agent.py

```diff
@@ -61,7 +62,44 @@ def __init__(self, config_path, **kwargs):
         self.config = utils.load_config(config_path)
         self.group_agent = {}
         self._connection = None
-
+        self.publish_settings = self.config.get('publish-settings')
+        self._remote_agent = None
+
+        if self.publish_settings:
+            self.publish_local = self.publish_settings.get('publish-local', True)
+            self.publish_remote = self.publish_settings.get('publish-remote', False)
+            remote = self.publish_settings.get('remote')
+            if self.publish_remote and not remote:
+                raise ValueError("Configured publish-remote without remote section")
+
+            self.remote_identity = remote.get('identity', None)
+            self.remote_serverkey = remote.get('serverkey', None)
+            self.remote_address = remote.get('vip-address', None)
+
+            if self.publish_remote:
+                assert self.remote_identity
+                assert self.remote_serverkey
+                assert self.remote_address
+
+    @property
+    def remote_agent(self):
+        if self._remote_agent is None:
```

Comment:
If we create the remote agent only once, why not create the agent when reading config and pass the remote agent to AlertGroup?

Reply:
because if the agent disconnects or doesn't start up then the lower level group agent doesn't know how to connect.  Also, there are many AgentGroup gevent processes that all share the main agent.

Reply:
Reconnect happens if the very first ping to the remote instance fails, but if the connection fails after the first successful alert then it won't reconnect.

Reply:
Fixed!